public/app/features/logs/components/panel/links.ts

```diff
@@ -0,0 +1,61 @@
+import { LinkModel } from '@grafana/data';
+
+import { FieldDef } from '../logParser';
+
+export function getTraceFromLinks(fields: FieldDef[]) {
+  for (const field of fields) {
+    if (!field.links) {
+      continue;
+    }
+    for (const link of field.links) {
+      const trace = getTraceFromLink(link);
+      if (trace) {
+        return trace;
+      }
+    }
+  }
+  return null;
+}
+
+function getTraceFromLink(link: LinkModel) {
+  const queryData = getDataSourceAndQueryFromLink(link);
```

Comment:
Lots of data sources have multiple derived fields, often one using regex and another using label/metadata, there's probably no way to grab the "right" one if we grab the first traceQL link?